src/blocks/shared/utils/date.ts

```diff
@@ -1,7 +1,33 @@
-export const SHORT_DATE_FORMAT: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric', year: 'numeric' };
-export const SHORT_DATE_FORMAT_WITHOUT_YEAR: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric' };
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+const SHORT_DATE_FORMAT: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric', year: 'numeric' };
+const SHORT_DATE_FORMAT_WITHOUT_YEAR: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric' };
+const DATE_NOT_AVAILABLE_MESSAGE = __( 'Date N/A', 'wp-parsely' );
+
+/**
+ * Returns whether the passed date can be processed further.
+ *
+ * @param {Date} date The date to be examined.
+ * @return {boolean} Whether the date can be processed further.
+ */
+export function canProcessDate( date: Date|string ): boolean {
+	if ( 'string' === typeof date ) {
+		date = new Date( date );
+	}
+
+	// Return false if the object is not a valid Date object, or if its value is
+	// equal to the Unix Epoch.
+	return date instanceof Date && ! isNaN( +date ) && 0 !== date.getTime();
```

Comment:
What is the purpose of this expression `0 !== date.getTime()`?

Reply:
`0 === date.getTime()` would mean that the passed date is the Unix Epoch date, as mentioned in the second part of the comment above. We could break the expression into 2 vars (`is_valid_date_object` and `$is_unix_epoch`) if you think it is needed, but I find that the comment is adequately explanatory.

Reply:
Recommendation of breaking the expression seems like a more easy to understand approach but I will leave the decision of adding it or not to you :)

Reply:
Ah, just saw your comment. Let me do this before you re-review.